R/multitables.R

```diff
@@ -26,6 +26,44 @@ setMethod("[[", c("MultitableCatalog", "numeric"), function (x, i, ...) {
     getEntity(x, i, Multitable, ...)
 })
 
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "character", "missing", "formula"),
+          function (x, i, j, value) {
+              stopifnot(length(i) == 1)
+              
+              ds <- loadDataset(datasetReference(x))
+              template <- formulaToQuery(value, data = ds)
+              
+              if (i %in% names(x)) {
+                  payload <- makeMultitablePayload(template = template)
+                  crPATCH(urls(x)[match(i, names(x))], body=toJSON(payload))
+                  return(x)
+              } else {
+                  ## Creating a new filter
+                  f <- .newMultitable(template, self(x), name = i)
+                  return(refresh(x))
+              }
+          })
+
+makeMultitablePayload <- function (template, ...) {
+    template <- lapply(template$dimensions, function (x) list(query=x))
+
+    return(wrapEntity(template = template, ...))
+}
+
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "ANY", "missing", "NULL"),
+          function (x, i, j, value) {
+              stopifnot(length(i) == 1)
+              if (i %in% names(x) | is.numeric(i)) {
+                  if (is.null(value)) {
```

Comment:
Maybe I missed your answer in the various commits, but why is the is.null check needed here? Doesn't method dispatch assure you that value is NULL?